### PR TITLE
feat(eg/source): add a new data source to query all kinds of the event sources

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add a new data source to query all kinds of the event sources.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
Add a new data source to query all kinds of the event sources.
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run=TestAccDataEventSources_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run=TestAccDataEventSources_basic -timeout 360m -parallel 4
=== RUN   TestAccDataEventSources_basic
--- PASS: TestAccDataEventSources_basic (133.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        133.276s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/data-sources/eg_event_sources.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region where the event sources are located.  
+  If omitted, the provider-level region will be used.
+
+* `provider_type` - (Optional, String) Specifies the type of the event sources to be queried.
```
Comment (deer-hang) [inline docs/data-sources/eg_event_sources.md]: Please list the enumeration values ​​for this field.

File: docs/data-sources/eg_event_sources.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+* `provider_type` - The provider type of the event source.
+
+* `event_types` - The event types that official event source provided.
+The [event_types](#eg_source_event_types) structure is documented below.
+
+* `type` - The type of the event source.
+
+* `description` - The description of the event source.
+
+* `status` - The status of the event source.
```
Comment (deer-hang) [inline docs/data-sources/eg_event_sources.md:91]: Please list the enumeration values ​​for this field.

File: huaweicloud/services/acceptance/eg/data_source_huaweicloud_eg_event_sources_test.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+}
+`, acceptance.HW_DMS_ROCKETMQ_INSTANCE_ID, name)
+}
+
+func testAccDataEventSources_basic_step1(name string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_eg_custom_event_source" "test" {
+  channel_id = huaweicloud_eg_custom_event_channel.test.id
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/eg/data_source_huaweicloud_eg_event_sources_test.go:139]: Aligning fields of scripts.

Comment (deer-hang) [inline huaweicloud/services/acceptance/eg/data_source_huaweicloud_eg_event_sources_test.go]:
```suggestion
  channel_id  = huaweicloud_eg_custom_event_channel.test.id
```
Aligning fields of scripts.
Same to line 161

File: docs/data-sources/eg_event_sources.md (hunk 1 above)
Comment (Lance52259) [reply docs/data-sources/eg_event_sources.md]: all enums are supported.

File: docs/data-sources/eg_event_sources.md (hunk 2 above)
Comment (Lance52259) [reply docs/data-sources/eg_event_sources.md:91]: done.

File: huaweicloud/services/acceptance/eg/data_source_huaweicloud_eg_event_sources_test.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/eg/data_source_huaweicloud_eg_event_sources_test.go]: all be done.

**Pull request merged**